src/main/java/org/apache/solr/benchmarks/indexing/DocReader.java

```diff
@@ -0,0 +1,17 @@
+package org.apache.solr.benchmarks.indexing;
+
+import java.io.IOException;
+import java.util.List;
+
+public interface DocReader extends AutoCloseable {
```

Comment:
This is actually not a `DocReader`. This is agnostic of docs. it is a `BatchedLinesReader`

Reply:
Thank you for bringing that up!

I named it `DocReader` as it's specific to our usage (reading docs as lines of string) even though it looks pretty much like a `BatchedLinesReader` as u mention. The implementing class does expect to read in lines and perform any pre-processing (remove `_version_` for example) such that the line read are ready to be consumed as Doc as is.

Probably cannot just use `Iterator` as it's a bit too generic, code reader could be hard to reason what it exactly is and this interface does more than just iteration (for example reading a batch of input docs) and could have extra methods added for other doc specific operations in the future :)